File: Semester 1/Class15_Computer_Contest_Lv_2_Notes_and_Homework_Dec_18/lvl2_hw_sol/lvl2_uneven_sand.cpp
#include <iostream>

int main() {

    int l=1,r=2e9;
    while(l<=r){
        int m=l+(r-l)/2;
        std::cout<<m<<std::endl;
        std::string s;std::cin>>s;
        std::cout.flush();
        if(s=="SINKS")l=m+1;
        else if(s=="FLOATS")r=m-1;
        else break;
    }
}

File: Semester 1/Class15_Computer_Contest_Lv_2_Notes_and_Homework_Dec_18/lvl2_hw_sol/lvl2_uneven_sand.py
import sys
input = sys.stdin.readline
lo, hi = 1, int(2e9)
while lo <= hi:
    mid = (lo + hi)//2
    print(mid)
    sys.stdout.flush()
    s = input().strip()
    if s == 'FLOATS':
        hi = mid - 1
    elif s == 'SINKS':
        lo = mid + 1
    else:
        break